packages/router/src/router.tsx

```diff
@@ -151,51 +139,112 @@ const Router: React.FC<RouterProps> = ({
   </LocationProvider>
 )
 
-const LocationAwareRouter: React.FC<RouterProps> = ({
+/**
+ * `namedRoutes` is the `routes` object that you import from `@redwoodjs/router`:
+ *
+ * ```js
+ * import { Link, routes } from '@redwoodjs/router'
+ *
+ * // ...
+ *
+ * <Link to={routes.home()}>Home</Link>
+ *
+ * // ...
+ * ```
+ *
+ * It's populated by the router, at run-time.
+ */
+const namedRoutes: AvailableRoutes = {}
+
+const LocationAwareRouter = ({
   useAuth,
   paramTypes,
   pageLoadingDelay,
   children,
-}) => {
+}: React.PropsWithChildren<RouterProps>) => {
   const location = useLocation()
+
   const flatChildArray = flattenAll(children)
 
-  const hasHomeRoute = flatChildArray.some((child) => {
-    if (isRoute(child)) {
-      return child.props.path === '/'
-    }
+  /**
+   * Of the router's children that are routes but aren't the not-found route
+   * (so routes and redirect routes), validate their props.
+   *
+   * @remarks
+   *
+   * Routes need to have `path`, `page`, and `name`,
+   * while redirect routes need to have `path` and `redirect`:
+   *
+   * ```js
+   * <Route path="/" page={HomePage} name="home" />  // ok
+   * <Route path="/" redirect="/home" />  // ok
+   * ```
+   */
+  flatChildArray
+    .filter((child) => isRoute(child) && !child.props.notfound)
+    .forEach((child) => {
+      const { path, redirect, page, name } = (
+        child as React.Component<RouteProps & RedirectRouteProps>
```

Comment:
Another type assertion to illustrate what should only be in the array at this point.